### PR TITLE
Support labels in docker build

### PR DESCRIPTION
**Pull request by hephex**
This change adds two new params to pass a list of labels to `docker build`. It will be possible to pass them as a YAML map using the parameter `labels`, or as JSON file containing (like `build_args`). The new parameters will allow setting some labels at build time without changing the Dockerfile.

Let me know if you think it's useful and if any change is needed :)

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@hephex Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=concourse/docker-image-resource&pullRequestId=208)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=concourse/docker-image-resource&pullRequestId=208) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@hephex Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal?repositoryId=concourse/docker-image-resource&pullRequestId=208)!

File: assets/out [hunk 1]
```diff
@@ -186,6 +188,26 @@ elif [ -n "$build" ]; then
...
+    done
+  fi
+
+  if [ -n "$labels_file" ]; then
+    labels_keys=($(jq -r 'keys | join(" ")' "$labels_file"))
+    if [ "${#labels_keys[@]}" -gt 0 ]; then
+      for key in "${labels_keys[@]}"; do
+        value=$(jq -r --arg "k" "$key" '.[$k]' "$labels_file")
+        expanded_build_args+=("--label")
+        expanded_build_args+=("${key}=${value}")
```
Comment (hephex) [inline assets/out]: I reused `expanded_build_arg` to pass the variables, should the labels have their own variable? e.g. `expanded_labels`

Comment (vito) [reply assets/out]: Maybe we could just give it a more general name? Looks like the `target` var could also just use the same var.

**Comment by vito** [comment]
Hey, sorry it's taken me so long to get to this! Thanks for submitting it.

If you're still around, I'd be happy to merge this once it's rebased, and if you want to rename the var you mentioned that'd be welcome too. :+1: Thanks for adding tests and everything!

**Comment by chrishiestand** [comment]
@hephex I'd like to see this merged as well. Let me know if you'd like help with the rebase or anything else.

**Comment by hephex** [comment]
sorry I didn't check my github account for a while, I'll rebase and make that change

**Comment by chrishiestand** [comment]
@vito I don't want to pester you, but when you can merge (or delegate the merge of) this PR I would start using this feature right away if it LGTY.

**Comment by vito** [comment]
Woops didn't notice this! Lgtm, thanks!

**Pull request merged**